build/scripts/build.ps1

```diff
@@ -1,84 +1,401 @@
+
+    $corerun = Join-Path $unitDir "CoreRun.exe"
+    $args = Join-Path $unitDir "xunit.console.netcore.exe"
+    foreach ($dll in Get-ChildItem -re -in "*.UnitTests.dll" $unitDir) {
+        $args += " $dll";
+    }
+
+    $args += " -parallel all"
+    $args += " -xml $logFile"
+
+    Write-Host "Running CoreClr tests"
+    Exec-Command $corerun $args | Out-Host
+}
+
+# Core function for running our unit / integration tests tests
+function Test-XUnit() { 
+
+    if ($testCoreClr) {
+        Test-XUnitCoreClr
+        return
+    }
+
+    # To help the VS SDK team track down their issues around install via build temporarily 
+    # re-enabling the build based deployment
+    # 
+    # https://github.com/dotnet/roslyn/issues/17456
+    $deployExtensionViaBuild = $false
+
+    if ($build) {
+        $deployArg = if ($deployExtensionViaBuild) { "true" } else { "false" }
+        Run-MSBuild "Roslyn.sln /p:Configuration=$buildConfiguration /p:DeployExtension=$deployArg"
+        
+        if ($testDesktop) { 
+            Run-MSBuild "src\Samples\Samples.sln /p:Configuration=$buildConfiguration /p:DeployExtension=false"
+        }
+
+        Stop-BuildProcesses
+    }
+
+    $anyVsi = $testVsi -or $TestVsiNetCore
+    if ($anyVsi -and (-not $deployExtensionViaBuild)) {
+        Deploy-VsixViaTool
+    }
+
+    $unitDir = Join-Path $configDir "UnitTests"
+    $runTests = Join-Path $configDir "Exes\RunTests\RunTests.exe"
+    $xunitDir = Join-Path (Get-PackageDir "xunit.runner.console") "tools"
+    $args = "$xunitDir"
 
-    $nuget = Ensure-NuGet
-    if ($clearPackageCache) {
-        Clear-PackageCache
+    if ($testDesktop) {
+        if ($test32) {
+            $dlls = Get-ChildItem -re -in "*.UnitTests.dll" $unitDir
+        }
+        else {
+            $dlls = Get-ChildItem -re -in "*.UnitTests.dll" -ex "*Roslyn.Interactive*" $unitDir 
+        }
     }
+    elseif ($testVsi) {
+        $dlls = Get-ChildItem -re -in "*.IntegrationTests.dll" $unitDir
+    }
+    else {
+        $dlls = Get-ChildItem -re -in "*.IntegrationTests.dll" $unitDir
+        $args += " -trait:Feature=NetCore"
+    }
+
+    if ($cibuild) {
+        $args += " -xml -timeout:50"
+
+        $procdumpPath = Ensure-ProcDump
+        $args += " -procdumppath:$procDumpPath"
+    }
+
+    if ($test64) {
+        $args += " -test64"
+    }
+
+    foreach ($dll in $dlls) { 
+        $args += " $dll"
+    }
+    
+    try {
+        Exec-Command $runTests $args | Out-Host 
+    }
+    finally {
+        Get-Process "xunit*" -ErrorAction SilentlyContinue | Stop-Process    
+    }
+}
+
+# Deploy our core VSIX libraries to Visual Studio via the Roslyn VSIX tool.  This is an alternative to 
+# deploying at build time.
+function Deploy-VsixViaTool() { 
+
+    $vsixDir = Get-PackageDir "roslyntools.microsoft.vsixexpinstaller"
+    $vsixExe = Join-Path $vsixDir "tools\VsixExpInstaller.exe"
+    $vsDir = [IO.Path]::GetFullPath("$msbuildDir\..\..\..\").Trim("\")
```

Comment:
Yep and I explicitly avoid it because of the number of times it's caused hours of head to desk problems. 

Resolve-Path returns a `PathInfo` object, not a `string`. Yet it's used in places that almost always want strings. In general this doesn't affect the outcome. But when it does the effects are subtle and result in a lot of frustration.

`GetFulPath` does the right thing 100% of the time. No frustration needed